### ISSUE TITLE
[CI] SharedClusterSnapshotRestoreIT testCloseOrDeleteIndexDuringSnapshot failed

### ISSUE THREAD

**Issue by droberts195** [elastic/elasticsearch#39828]
`org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT testCloseOrDeleteIndexDuringSnapshot` failed in https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+matrix-java-periodic/ES_BUILD_JAVA=java11,ES_RUNTIME_JAVA=java11,nodes=immutable&&linux&&docker/60/console

The REPRO command was:

```
./gradlew :server:integTest \
  -Dtests.seed=BBE286746EB2E3EC \
  -Dtests.class=org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT \
  -Dtests.method="testCloseOrDeleteIndexDuringSnapshot" \
  -Dtests.security.manager=true \
  -Dtests.locale=fr-CH \
  -Dtests.timezone=America/Ojinaga \
  -Dcompiler.java=11 \
  -Druntime.java=11
```

This didn't reproduce locally for me.

The exception was:

```
FAILURE 2.98s J1 | SharedClusterSnapshotRestoreIT.testCloseOrDeleteIndexDuringSnapshot <<< FAILURES!
   > Throwable #1: java.lang.AssertionError: 
   > Expected: <SUCCESS>
   >      but: was <PARTIAL>
   > 	at __randomizedtesting.SeedInfo.seed([BBE286746EB2E3EC:A0CB08C85FC687F8]:0)
   > 	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
   > 	at org.elasticsearch.snapshots.SharedClusterSnapshotRestoreIT.testCloseOrDeleteIndexDuringSnapshot(SharedClusterSnapshotRestoreIT.java:2529)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
   > 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
   > 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
   > 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
   > 	at java.base/java.lang.Thread.run(Thread.java:834)
```

**Comment by droberts195** [comment]
This failed in the same way in another 7.x build: https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+release-tests/114/console

I muted this test for 7.x in 366eef99a18563b7b4cff19b00ffcf71a2d4977e

**Comment by ywelsch** [comment]
@original-brownbear any thoughts on this one?

**Comment by original-brownbear** [comment]
@ywelsch sorry for letting this one fall off of the radar ...

I tried reproducing this back in the day when it was raised but couldn't. I also couldn't find any new occurrences of this in `master` (where it's not muted). Given that between now and the time this failed there were a few backports of snapshot fixes to 7.x and both code bases should be in sync now, maybe just unmute this test again in 7.x?

**Comment by ywelsch** [comment]
ok, let's unmute in 7.x

**Comment by davidkyle** [comment]
@original-brownbear here's the 7.x failure you were waiting for 

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+7.x+multijob-unix-compatibility/os=opensuse/157/console

Same error as above

**Comment by original-brownbear** [comment]
@davidkyle ah thanks for pinging. Looking into it now

**Comment by original-brownbear** [comment]
Seed `889B596377F8BCF2` reproduces 100% of the time here for me.

**Comment by original-brownbear** [comment]
closed via ea7db2b
